pkg/kv/kvserver/replicastats/replica_stats.go

```diff
@@ -89,15 +89,15 @@ type ReplicaStats struct {
 }
 
 type replicaStatsRecord struct {
-	localityCounts PerLocalityCounts
+	localityCounts *PerLocalityCounts
```

Comment:
@kvoli `PerLocalityCounts` is a `map`, whose header is already a represented by a pointer. So this change is introducing an extra level of indirection and forcing this map header into a separate, 8 byte heap allocation. That's fine, but is it necessary for this change? Could we have just lazily allocated the hash map and compared the map header against nil?

See https://go.dev/play/p/mbePNNDnHji.

Reply:
Certainly not necessary if the map were just left uninitialized when not used. We could have just compared the map against nil as you mentioned.